index.js

```diff
@@ -61,17 +78,23 @@ app.use(async (ctx) => {
     ctx.status = 405;
     return;
   }
-
+  
   if (ctx.ws) {
     const ws = new WebSocketJSONStream(await ctx.ws());
-    db.listen(ws, docId);
+    db.listen(ws, docId); // docId is passed to 'connect' middleware as ctx.req
   } else {
-    ctx.body = 'Document exists.';
+    ctx.body = {docId, readOnly};
   }
 });
 
 app.listen(process.env.PORT || 8080);
 
-function isEmptyObject(obj) {
-  return Object.keys(obj).length === 0 && obj.constructor === Object;
+function getSessionDetails(sessionId) {
+  for (let session of documents) {
```

Comment:
`session` does not seem to be reassigned here – let's prefer `const` as opposed to `let` for its declaration.